dev/tests/api-functional/testsuite/Magento/Search/Api/SearchTest.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Search\Api;
+
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Framework\Webapi\Rest\Request;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\WebapiAbstract;
+
+class SearchTest extends WebapiAbstract
+{
+    const SERVICE_VERSION = 'V1';
+    const SERVICE_NAME = 'searchV1';
+    const RESOURCE_PATH = '/V1/search/';
+
+    /**
+     * @var ProductInterface
+     */
+    private $product;
+
+    protected function setUp()
+    {
+        $productSku = 'simple';
+
+        $objectManager = Bootstrap::getObjectManager();
+        $productRepository = $objectManager->create(ProductRepositoryInterface::class);
+        $this->product = $productRepository->get($productSku);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Catalog/_files/product_simple.php
+     */
+    public function testExistingProductSearch()
+    {
+        $productName = $this->product->getName();
+
+        $searchCriteria = $this->buildSearchCriteria($productName);
+        $serviceInfo = $this->buildServiceInfo($searchCriteria);
+
+        $response = $this->_webApiCall($serviceInfo, $searchCriteria);
+
+        self::assertArrayHasKey('search_criteria', $response);
+        self::assertArrayHasKey('items', $response);
+        self::assertGreaterThan(0, count($response['items']));
+        self::assertGreaterThan(0, $response['items'][0]['id']);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Catalog/_files/product_simple.php
```

Comment:
You need in this case the @magentoApiDataFixture with `product_simple` ?

Reply:
Good point :) We need it since in case with no products the response from the request is completely different. The idea is to test the search request with some products in the database

Reply:
That's clear now for me 😄